util/flocks/src/frwlock/write_guard.rs

```diff
@@ -0,0 +1,31 @@
+use std::ops::{Deref, DerefMut};
+use rustix::{
+    fs::{flock, FlockOperation},
+    fd::AsFd
+};
+
+pub struct FrwLockWriteGuard<T: AsFd> {
+    pub(crate) data: *mut T
+}
```

Comment:
Also here, I think we're just hiding the unsafety creating a massive source of undefined behavior.

Reply:
Same as above.